addon/bootstrap.js

```diff
@@ -1,17 +1,65 @@
-/* globals Components, AddonManager */
+/* globals AddonManager, Components, Services */
 /* eslint-disable no-unused-vars */
 
 const OLD_ADDON_PREF_NAME = "extensions.jid1-NeEaf3sAHdKHPA@jetpack.deviceIdInfo";
 const OLD_ADDON_ID = "jid1-NeEaf3sAHdKHPA@jetpack";
+const ADDON_ID = "pageshot@mozilla.org";
+const TELEMETRY_PREF = "toolkit.telemetry.enabled";
+const USER_ENABLE_PREF = "extensions.pageshot.userdisabled";
+const SYSTEM_ENABLE_PREF = "extensions.pageshot.systemdisabled";
 
-const prefs = Components.classes["@mozilla.org/preferences-service;1"]
-  .getService(Components.interfaces.nsIPrefBranch);
-Components.utils.import("resource://gre/modules/AddonManager.jsm");
+const { interfaces: Ci, utils: Cu } = Components;
+Cu.import("resource://gre/modules/AddonManager.jsm");
+Cu.import("resource://gre/modules/Console.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+const { EmbeddedExtensionManager } = Cu.import("resource://gre/modules/LegacyExtensionsUtils.jsm");
+
+const prefs = Services.prefs;
+
+function shouldStartWebExtension() {
+  return ! (
+    prefs.prefHasUserValue(USER_ENABLE_PREF) &&
+    prefs.getBoolPref(USER_ENABLE_PREF)
+  ) || (
+    prefs.getPrefType(SYSTEM_ENABLE_PREF) &&
+    prefs.getBoolPref(SYSTEM_ENABLE_PREF)
+  );
+}
```

Comment:
This is probably my fault too, but the expression here is incorrect.  I think it's `(! USER_ENABLE_PREF) || SYSTEM_ENABLE_PREF` – as a result setting `extensions.pageshot.systemdisabled` to true forces Page Shot *on* rather than off.